tests/system/src/test/java/test/robot/javafx/stage/StageFocusTest.java

```diff
@@ -0,0 +1,116 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.stage;
+
+import java.util.concurrent.CountDownLatch;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.input.KeyCode;
+import javafx.scene.paint.Color;
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+import test.util.Util;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class StageFocusTest {
+
+    static CountDownLatch startupLatch;
+
+    static final double STAGE_SIZE = 200;
+
+    static final double STAGE_X = 100;
+    static final double STAGE_Y = 100;
+
+    static boolean receivedEvent = false;
+
+    Robot robot;
+
+    public static class TestApp extends Application {
+        @Override
+        public void start(Stage stage) {
+            Platform.setImplicitExit(false);
+
+            Group root = new Group();
+            Scene scene = new Scene(root, STAGE_SIZE, STAGE_SIZE);
+            scene.setFill(Color.LIGHTGREEN);
+            scene.setOnKeyPressed(e -> {
+                if (e.getCode() == KeyCode.A)
+                    receivedEvent = true;
+            });
+
+            stage.setScene(scene);
+            stage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> {
+                Platform.runLater(() -> {
+                    stage.setX(STAGE_X);
+                    stage.setY(STAGE_Y);
+                    startupLatch.countDown();
+                });
+            });
+            stage.show();
+        }
+    }
+
+    @BeforeAll
+    public static void setupOnce() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        Util.launch(startupLatch, TestApp.class);
+    }
+
+    @AfterAll
+    public static void teardown() {
+        Util.shutdown();
+    }
+
+    @BeforeEach
+    public void setup() {
+        Util.runAndWait(() -> robot = new Robot());
+    }
+
+    /**
+     * Checks whether Stage is actually shown when calling show()
+     *
+     * Meant as a "canary" test of sorts to ensure other tests relying on
+     * Stage being actually shown and on foreground work fine.
+     */
+    @Test
+    public void testStageHasFocusAfterShow() {
+        Util.sleep(250);
+        Util.runAndWait(() -> {
+            robot.keyPress(KeyCode.A);
+        });
+        assertTrue(receivedEvent, "Expected key press has NOT been received! Stage did not have focus after showing. Some tests might fail because of this." +
```

Comment:
Do you need to sleep before the assertion to ensure that the event has been delivered?